src/client/Microsoft.Identity.Client/RegionDiscoveryOutcome.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Identity.Client.Region;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Contains the result of region when MSAL region discovery is used, 
+    /// published as part of AuthenticationResultMetadata.
+    /// <see cref="AuthenticationResultMetadata"/> for additional metadata 
+    /// information of the authentication result.
+    /// </summary>
+    public class RegionDiscoveryOutcome
+    {
+        /// <summary>
+        /// Constructor for RegionDiscoveryOutcome
+        /// </summary>
+        /// <param name="regionOutcome"></param>
+        /// <param name="regionUsed"></param>
+        /// <param name="exception"></param>
+        internal RegionDiscoveryOutcome(
+            RegionOutcome regionOutcome,
+            string regionUsed,
+            string exception)
+        {
+            RegionOutcome = regionOutcome;
+            RegionUsed = regionUsed;
+            Exception = exception;
+        }
+
+        /// <summary>
+        /// Region Outcome based on MSAL region detection
+        /// </summary>
+        internal RegionOutcome RegionOutcome { get; }
+
+        /// <summary>
+        /// Region returned to the user
+        /// </summary>
+        internal string RegionUsed { get; }
```

Comment:
Wondering if we should output RegionDetected as well.

Reply:
Hmm, I think it would be helpful. I just remembered that we talked about this some time ago (#2698) and even started work on that (#2684).
Maybe it would be better to have only `RegionDetected` without `RegionUsed`?
- If a user passes in a region, since it's the one that is used, they already know it. So `RegionDetected` here is more helpful to know if region passed in is correct or not; `RegionUsed` is redundant.
- If user selected auto-detection, then `RegionDetected` is more helpful, and is the same as `RegionUsed`, which is again redundant.

Reply:
RegionUsed already has the detected region when autodetection is requested. I added a telemetry property for this instead